### ISSUE TITLE
Binary is missing subtype

### ISSUE THREAD

**Issue by rueckstiess** [mongodb-js/extended-json#11]
In https://github.com/mongodb-js/extended-json/blob/master/lib/types.js#L15-19:

```
    Binary: function(v) {
      return {
        '$binary': v.buffer.toString('base64')
      };
    },
```

But `$binary` requires a `$type` argument: http://docs.mongodb.org/manual/reference/mongodb-extended-json/#binary

**Comment by efmr** [comment]
Something like this would solve the issue, right?

```javascript
//inflate
Binary: function(v) {
  return {
    '$binary': v.buffer.toString('base64'),
    '$type': v.sub_type.toString(16) //shellmode: '$type': v.sub_type
  };
},
//deflate
'$binary': function(val) {
  return bson.Binary(new Buffer(val.$binary, 'base64'), parseInt(val.$type, 16));  //shellmode: parseInt(val.$type)
},
```

**Comment by rueckstiess** [comment]
:+1: I think that would do.

**Comment by rueckstiess** [comment]
Because neither shell nor log mode are actual parsable json, we need to regex.replace() the strings beforehand instead of using a `reviver` function on `EJSON.parse()`. On the way out, however, we could do either:

1. Use a `replacer` function in `EJSON.stringify()`, see [here](http://mdn.io/JSON.stringify#The_replacer_parameter)
2. `.stringify()` as usual and then regex again to go from strict to shell or log

**Comment by imlucas** [comment]
published in v1.5.2 

thanks again @efmr!
